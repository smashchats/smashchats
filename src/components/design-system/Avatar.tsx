import React, { type PropsWithChildren } from "react";
import { View } from "react-native";

type Props = PropsWithChildren<{
    flex?: number;
    bg?: string;
    bgColor?: string;
    borderRadius?: number;
    size?: number;
}>;

export function Avatar({
    children,
    flex,
    bg,
    bgColor,
    borderRadius,
    size,
}: Props): JSX.Element {
    return (
        <View
            style={{
                backgroundColor: bg ?? bgColor,
                flex,
                minWidth: size ?? 64,
                maxWidth: size ?? 64,
                width: size ?? 64,
                minHeight: size ?? 64,
                maxHeight: size ?? 64,
                height: size ?? 64,
                borderRadius: borderRadius ?? 16,
                overflow: "hidden",
            }}
        >
            {children}
        </View>
    );
}
