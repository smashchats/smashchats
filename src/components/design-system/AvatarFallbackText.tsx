import React from "react";
import { Text, View } from "react-native";

type Props = {
    name: string;
};

export function AvatarFallbackText({ name }: Readonly<Props>): JSX.Element {
    return (
        <View
            style={{
                alignItems: "center",
                justifyContent: "center",
                height: "100%",
                width: "100%",
            }}
        >
            <Text style={{ fontSize: 24, color: "white", fontWeight: "bold" }}>
                {(name ?? "")
                    .split(" ")
                    .map((word) => word[0])
                    .join("")
                    .toUpperCase()}
            </Text>
        </View>
    );
}
