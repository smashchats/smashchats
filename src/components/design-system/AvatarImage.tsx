import React, { type PropsWithChildren } from "react";
import { ViewStyle } from "react-native";
import { Image, ImageSource } from "expo-image";

type Props = PropsWithChildren<
    {
        source?:
            | ImageSource
            | string
            | number
            | ImageSource[]
            | string[]
            | null;
        alt?: string;
        size?: number;
    } & ViewStyle
>;

export function AvatarImage({ source, alt, size }: Props): JSX.Element {
    return (
        <Image
            style={{
                minWidth: size ?? 64,
                maxWidth: size ?? 64,
                width: size ?? 64,
                minHeight: size ?? 64,
                maxHeight: size ?? 64,
                height: size ?? 64,
            }}
            source={source}
            alt={alt}
            contentFit={"contain"}
            transition={1000}
        />
    );
}
