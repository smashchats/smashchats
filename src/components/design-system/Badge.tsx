import { Colors } from "@/src/constants/Colors.js";

import React, { type PropsWithChildren } from "react";
import { DimensionValue, View, StyleSheet, ViewStyle } from "react-native";

type Props = PropsWithChildren<
    {
        flex?: number;
        type?: string;
        bg?: string;
        bgColor?: string;
        size?: string;
        h?: DimensionValue;
    } & ViewStyle
>;

export function Badge({
    children,
    size,
    flex,
    bg,
    h,
    width,
    marginHorizontal,
    marginRight,
    paddingHorizontal,
    type,
    borderRadius,
    borderColor,
    bgColor,
    borderWidth,
    ...rest
}: Props): JSX.Element {
    return (
        <View
            style={{
                flexDirection: "row",
                alignItems: "center",
                justifyContent: "center",
                borderRadius: 16,
                paddingHorizontal: 8,
                paddingVertical: 1,

                height: h,
                flex,
                width,
                marginHorizontal,
                marginRight,

                ...styles.unselected,
                ...(type === "unselected" ? styles.unselected : {}),
                ...(type === "trusted" ? styles.trusted : {}),
                ...(type === "smashed" ? styles.smashed : {}),

                ...(borderWidth && { borderWidth }),
                ...(borderColor && { borderColor }),
                ...((bg || bgColor) && { backgroundColor: bg ?? bgColor }),
                ...(borderRadius && { borderRadius }),
                ...rest,
            }}
        >
            {children}
        </View>
    );
}

const styles = StyleSheet.create({
    smashed: {
        backgroundColor: Colors.purple,
        borderColor: Colors.purple,
        borderWidth: 2,
        // marginLeft: 30
    },
    trusted: {
        backgroundColor: "#3f3f3f",
        borderColor: "#3f3f3f",
        borderWidth: 2,
    },
    unselected: {
        backgroundColor: Colors.background,
        borderColor: Colors.purple,
        borderWidth: 2,
    },
});
