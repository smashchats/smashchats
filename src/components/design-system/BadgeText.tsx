import React, { type PropsWithChildren } from "react";
import { DimensionValue, Text, TextStyle, View } from "react-native";

type Props = PropsWithChildren<
    {
        bg?: string;
        h?: DimensionValue;
    } & TextStyle
>;

export function BadgeText({
    children,
    flex,
    bg,
    h,
    color,
    ...rest
}: Props): JSX.Element {
    return (
        <View style={{ backgroundColor: bg, height: h, flex }}>
            <Text
                style={{
                    color: color ?? "white",
                    fontFamily:
                        'Cairo, apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif',
                    fontSize: 12,
                    textTransform: "uppercase",
                    ...rest,
                }}
            >
                {children}
            </Text>
        </View>
    );
}
