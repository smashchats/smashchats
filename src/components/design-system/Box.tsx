import React, { type PropsWithChildren } from "react";
import { DimensionValue, View, ViewStyle } from "react-native";

type Props = PropsWithChildren<
    {
        bg?: string;
        h?: DimensionValue;
    } & ViewStyle
>;

export function Box({ children, bg, h, ...rest }: Props): JSX.Element {
    return (
        <View
            style={{
                backgroundColor: bg,
                height: h,
                ...rest,
            }}
        >
            {children}
        </View>
    );
}
