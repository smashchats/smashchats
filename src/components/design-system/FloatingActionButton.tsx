import React from "react";
import { Pressable, useWindowDimensions } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import { Colors } from "@/src/constants/Colors.js";
import { MaterialCommunityIcon } from "@/src/components/design-system/MaterialCommunityIconsType.js";

type Props = {
    icon: MaterialCommunityIcon;
    onPress: () => void;
};

export function FloatingActionButton({
    icon,
    onPress,
}: Readonly<Props>): JSX.Element {
    const { height } = useWindowDimensions();
    return (
        <Pressable
            testID="FloatingActionButton::Pressable"
            style={{
                zIndex: 999,
                width: 50,
                height: 50,
                position: "absolute",
                right: 0,
                top: height - 70,
                backgroundColor: Colors.purple,
                borderRadius: 25,
                marginRight: 20,
                marginBottom: 20,
                justifyContent: "center",
                alignItems: "center",
            }}
            onPress={() => {
                onPress();
            }}
        >
            <MaterialCommunityIcons name={icon} size={28} color={"white"} />
        </Pressable>
    );
}
