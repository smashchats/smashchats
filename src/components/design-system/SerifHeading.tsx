import React, { type PropsWithChildren } from "react";
import { TextStyle } from "react-native";

import { Heading } from "@/src/components/design-system/Heading.jsx";

type Props = PropsWithChildren<TextStyle>;

export function SerifHeading({ children, ...rest }: Props): JSX.Element {
    return (
        <Heading
            fontSize={20}
            style={{
                fontFamily: "Didot",
                fontSize: 20,
                marginVertical: 4,
                ...rest,
            }}
        >
            {children}
        </Heading>
    );
}
