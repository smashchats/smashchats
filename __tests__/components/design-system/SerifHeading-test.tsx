import { render } from "@testing-library/react-native";
import { SerifHeading } from "@/src/components/design-system/SerifHeading";

describe("SerifHeading", () => {
    test("Renders correctly", () => {
        const tree = render(
            <SerifHeading>
                Hello
            </SerifHeading>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
