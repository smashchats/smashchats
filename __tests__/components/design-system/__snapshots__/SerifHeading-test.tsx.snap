// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerifHeading Renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flex": undefined,
      "height": undefined,
      "marginBottom": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "Didot",
        "fontSize": 20,
        "fontWeight": undefined,
        "marginVertical": 4,
      }
    }
  >
    Hello
  </Text>
</View>
`;
